pyccl/baryons/baccoemu_baryons.py

```diff
@@ -176,3 +181,14 @@ def _include_baryonic_effects(self, cosmo, pk):
                     is_logp=pk.psp.is_log,
                     extrap_order_lok=pk.extrap_order_lok,
                     extrap_order_hik=pk.extrap_order_hik)
+
+    def _check_a_range(self, a):
+        if not isinstance(a, Iterable):
```

Comment:
We check for this using `np.ndim` in other parts of the code (https://github.com/LSSTDESC/CCL/blob/86125b08f66b27341e9bb67403f3f0ca3cece359/pyccl/halos/profiles/nfw.py#L202). Not that we need to be super homogeneous, but this would avoid importing Iterable, and I believe it works in the relevant cases.

Reply:
Done. Also replaced the `hasattr(*, "__len__")` while I was at it.